docs/User/Tutorials/module-structure-and-quick-examples.md

```diff
@@ -340,42 +361,66 @@ from stdin and to write a stream to stdout and stderr.
 
 Implementation of a console echo program:
 
-```haskell
+```haskell unshared
+import Data.Function ((&))
+import qualified Streamly.Console.Stdio as Stdio
+
 main =
-  Stdio.readChunks     -- Stream IO (Array Word8)
-    & Stdio.fromChunks -- IO ()
+  Stdio.readChunks    -- Stream IO (Array Word8)
+    & Stdio.putChunks -- IO ()
 ```
 
 An example to read two numbers from separate lines on stdin and sum them:
 
-```haskell
+```haskell unshared
+import Data.Function ((&))
+import qualified Streamly.Internal.Console.Stdio as Stdio
+import qualified Streamly.Data.Stream as Stream
+import qualified Streamly.Data.Fold as Fold
+
 main =
-    Stdio.readChars                           -- Stream IO Char
-      & Stream.splitOn (== '\n') Fold.toList  -- Stream IO String
-      & Stream.map read                       -- Stream IO Int
-      & Stream.take 2                         -- Stream IO Int
-      & Stream.fold Fold.sum                  -- IO Int
-      >>= print                               -- IO ()
+    Stdio.readChars                              -- Stream IO Char
+      & Stream.splitSepBy_ (== '\n') Fold.toList -- Stream IO String
+      & fmap read                                -- Stream IO Int
+      & Stream.take 2                            -- Stream IO Int
+      & Stream.fold Fold.sum                     -- IO Int
+      >>= print                                  -- IO ()
 ```
 
 ## File IO
 
 Implementation of the Unix `cp` utility to copy `input.txt` to `output.txt`:
 
-```haskell
-main =
-  File.readChunks (Path.fromString "input.txt")      -- Stream IO (Array Word8)
-    & File.fromChunks (Path.fromString "output.txt") -- IO ()
+```haskell unshared
+import Data.Function ((&))
+import qualified Streamly.FileSystem.FileIO as File
+import qualified Streamly.FileSystem.Path as Path
+import qualified Streamly.Data.Stream as Stream
+
+main = do
+  inp <- Path.fromString "input.txt"
+  out <- Path.fromString "output.txt"
```

Comment:
we can use the quasiquoter instead for static type check of these. That way we can use non-monadic code.